src/training_hub/algorithms/osft.py

```
@@ -156,70 +179,86 @@ def train(
+            raise ValueError("block_size required when is_pretraining=True")
 
         # Validate pretraining parameters
         if is_pretraining and unmask_messages:
             raise ValueError(
-                'Cannot use both is_pretraining=True and unmask_messages=True. These are mutually exclusive modes.'
+                "Cannot use both is_pretraining=True and unmask_messages=True. These are mutually exclusive modes."
             )
 
         if not is_pretraining and block_size is not None:
-            warnings.warn('block_size only valid with is_pretraining=True')
+            warnings.warn("block_size only valid with is_pretraining=True")
```

Ruff (0.14.14)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)